### PR TITLE
Update TopologySpec for v1alpha3 API

### PR DESCRIPTION
**Pull request by johngmyers**
I don't like that the values for `dns` are capitalized, whereas the others are not. In general, I'm not a fan of the capitalization, but is there a wider Kubernetes convention to follow?

### CODE REVIEW

File: pkg/apis/kops/v1alpha3/topology.go [hunk 1]
```diff
@@ -22,30 +22,27 @@ const (
 )
 
 type TopologySpec struct {
-	// The environment to launch the Kubernetes masters in public|private
-	Masters string `json:"masters,omitempty"`
+	// ControlPlane specifies the environment for launching the control plane nodes. (public, private)
+	ControlPlane string `json:"control-plane,omitempty"`
```
Comment (hakman) [inline pkg/apis/kops/v1alpha3/topology.go:26]: Not quite sure we have any other place using `-`.
```suggestion
	ControlPlane string `json:"controlPlane,omitempty"`
```

Comment (johngmyers) [reply pkg/apis/kops/v1alpha3/topology.go]: Fixed

**Review by hakman — APPROVED** [review]

**Comment by olemarkus** [comment]
> I don't like that the values for `dns` are capitalized, whereas the others are not. In general, I'm not a fan of the capitalization, but is there a wider Kubernetes convention to follow?

To answer this one: enum-type values are alaways capitalised (upper camel-case) in K8s APIs. I don't have a reference for that convention, but you see it more or less everywhere. So it's the other fields that are wrong. In v1alpha3, I think we should ensure the enums are typede typed and do conversions of the values so they get the correct case convention.

**Pull request merged**